rubicon/objc/api.py

```diff
@@ -403,12 +412,25 @@ def _objc_setter(objc_self, _cmd, new_value):
             if not isinstance(new_value, self.vartype):
                 # If vartype is a primitive, then new_value may be unboxed. If that is the case, box it manually.
                 new_value = self.vartype(new_value)
-            old_value = get_ivar(objc_self, '_' + attr_name)
-            if issubclass(self.vartype, objc_id) and new_value:
+
+            if self.weak:
+                old_value = get_weak_ivar(objc_self, '_' + attr_name)
+            else:
+                old_value = get_ivar(objc_self, '_' + attr_name)
```

Comment:
If `get_weak_ivar` and `get_ivar` were combined like I suggested in my other comment, this call could be simplified to

```suggestion
            old_value = get_ivar(objc_self, '_' + attr_name, weak=self.weak)
```

The same would apply to the `set_weak_ivar`/`set_ivar` calls below.